lib/hackney/income/gov_notify_gateway.rb

```diff
@@ -3,40 +3,76 @@
 module Hackney
   module Income
     class GovNotifyGateway
-      def initialize(sms_sender_id:, api_key:)
+      def initialize(api_host:, api_key:, sms_sender_id:)
+        @api_host = api_host
+        @api_key = api_key
         @sms_sender_id = sms_sender_id
-        @client = Notifications::Client.new(api_key)
       end
 
-      def send_text_message(phone_number:, template_id:, reference:, variables:)
-        @client.send_sms(
+      def send_text_message(tenancy_ref:, phone_number:, template_id:, reference:, variables:)
+        uri = URI("#{@api_host}/messages/send_sms")
+        body_data = {
+          tenancy_ref: tenancy_ref,
           phone_number: pre_release_phone_number(phone_number),
           template_id: template_id,
-          personalisation: variables,
+          variables: variables,
           reference: reference,
           sms_sender_id: @sms_sender_id
-        )
+        }.to_json
+
+        req = Net::HTTP::Post.new(uri)
+        req['X-Api-Key'] = @api_key
+        req['Content-Type'] = 'application/json'
+
+        res = Net::HTTP.start(uri.host, uri.port, use_ssl: true) { |http| http.request(req, body_data) }
+        unless res.is_a? Net::HTTPSuccess
+          raise Exceptions::IncomeApiError.new(res), 'error sending sms'
+        end
+
+        res
       end
 
-      def send_email(recipient:, template_id:, reference:, variables:)
-        @client.send_email(
+      def send_email(tenancy_ref:, recipient:, template_id:, reference:, variables:)
+        uri = URI("#{@api_host}/messages/send_email")
+        body_data = {
+          tenancy_ref: tenancy_ref,
           email_address: pre_release_email(recipient),
           template_id: template_id,
-          personalisation: variables,
+          variables: variables,
           reference: reference
-        )
+        }.to_json
+
+        req = Net::HTTP::Post.new(uri)
+        req['X-Api-Key'] = @api_key
+        req['Content-Type'] = 'application/json'
+        res = Net::HTTP.start(uri.host, uri.port, use_ssl: true) { |http| http.request(req, body_data) }
+        unless res.is_a? Net::HTTPSuccess
+          raise Exceptions::IncomeApiError.new(res), 'error sending email'
+        end
+
+        res
       end
 
       def get_text_templates
-        @client.get_all_templates(type: 'sms').collection.map do |template|
-          { id: template.id, name: template.name, body: template.body }
-        end
+        uri = URI("#{@api_host}/messages/get_templates")
+        uri.query = URI.encode_www_form('type' => 'sms')
+
+        req = Net::HTTP::Get.new(uri)
+        req['X-Api-Key'] = @api_key
+        res = Net::HTTP.start(uri.hostname, uri.port, use_ssl: true) { |http| http.request(req) }
+
```

Comment:
```suggestion
        unless response.is_a? Net::HTTPSuccess
          raise Exceptions::TenancyApiError.new(response), "when trying to get_text_templates '#{uri}'"
        end
```
add guards for non 200

Reply:
see test example at
https://github.com/LBHackney-IT/LBH-IncomeCollection/blob/master/spec/lib/hackney/income/search_tenancies_gateway_spec.rb#L154